$Id: 910_debian_Xserver_RTFF.diff 486 2005-08-03 04:51:11Z dnusinow $

Give the user a far stronger clue as to what to do when their font configuration
is horribly screwed up; this patch by Branden Robinson.

Index: a/dix/main.c
===================================================================
--- a/dix/main.c.orig	2006-11-13 19:59:22.000000000 +0100
+++ b/dix/main.c	2006-11-26 01:55:13.000000000 +0100
@@ -407,10 +407,42 @@
 			defaultFontPath);
 	}
 	if (!SetDefaultFont(defaultTextFont))
+#ifdef DEBIAN
+	    FatalError("could not open default font '%s';\n"
+"the X server's font paths might be misconfigured, remote font server(s)\n"
+"may be unreachable, and/or local fonts may not be installed or are not\n"
+"configured correctly.\n"
+"\n"
+"People inexperienced with the X Window System should have the\n"
+"\"xorg\" package installed.\n"
+"# apt-get install xorg\n"
+"\n"
+"Other useful commands to run include:\n"
+"$ dpkg --status x11-common\n"
+"$ dpkg --status xfonts-base\n"
+"$ zmore /usr/share/doc/x11-common/FAQ.gz", defaultTextFont);
+#else
 	    FatalError("could not open default font '%s'", defaultTextFont);
+#endif
 	if (!(rootCursor = CreateRootCursor(defaultCursorFont, 0)))
+#ifdef DEBIAN
+	    FatalError("could not open default cursor font '%s';\n"
+"the X server's font paths might be misconfigured, remote font server(s)\n"
+"may be unreachable, and/or local fonts may not be installed or are not\n"
+"configured correctly.\n"
+"\n"
+"People inexperienced with the X Window System should have the\n"
+"\"xorg\" package installed.\n"
+"# apt-get install xorg\n"
+"\n"
+"Other useful commands to run include:\n"
+"$ dpkg --status x11-common\n"
+"$ dpkg --status xfonts-base\n"
+"$ zmore /usr/share/doc/x11-common/FAQ.gz", defaultTextFont);
+#else
 	    FatalError("could not open default cursor font '%s'",
 		       defaultCursorFont);
+#endif
 #ifdef DPMSExtension
  	/* check all screens, looking for DPMS Capabilities */
  	DPMSCapableFlag = DPMSSupported();
